### ISSUE TITLE
Routing to WebSphere Application Server using an external router from OpenShift Origin

### ISSUE THREAD

**Issue by tienhngnguyen** [WASdev/ci.docker.websphere-traditional#87]
Hi,

I’m trying to route traffic from OpenShift Origin (like a nginx proxy) to the WebSphere Server which is basically running in a Docker container in OpenShift.
 
The URL to route the traffic to WebSphere looks like this: https://websphere-myproject.10.0.75.2.nip.io/ibm/console/login.do?action=secure, which actually arrives on WebSphere, however, WebSphere seems to give a wrong response (or routes me to another invalid URL because not having SSL enabled). How can I make the WebSphere only listens on that route/url so that it gives me a correct response that routes me to WebSphere Application Server Console (Port 9043)?
 
Basically. I’m using a router to route the traffic to the WebSphere Application Server, however, because the reason that WebSphere seems to control routing itself, I think there is a conflct when I try to manage the routing though, for example a HA PROXY, nginx-Server.
 
How can I solve this issue?

**Comment by PaddydudeIBM** [comment]
Did you ever manage to find a solution for this? I'm currently experiencing the same thing.

**Comment by tienhngnguyen** [comment]
Unfortunetly, Moor :/

PaddydudeIBM <notifications@github.com> schrieb am Di., 22. Okt. 2019,
15:25:

**Comment by ashokkumarangadi** [comment]
@tienhngnguyen @PaddydudeIBM : We are also facing same issue with WebSphere console, any workaround or solution for this???

**Comment by gituserJ** [comment]
WebSphere admin console was trying to load connecting to port 443 in OpenShift .Adding ports 80 & 443 in virtual host for admin host and redeploying the image resolved the issue.

**Comment by tienhngnguyen** [comment]
Hi,

after a long time, we have now made it work on GKE with Ingress, where we had the same issue. To solve this problem we had to add port 443 to the virtual host "admin_host"for the WebSphere console and add the following properties to the WebContainer: 
trusthostheaderport: true
com.ibm.ws.webcontainer.extractHostHeaderPort: true

This is because Ingress is only able to receive traffic on port 443 (secure port using https), which routes the traffic to WebSphere. Adding these properties to WebSphere will make sure that the actual target ports are extracted from the header so that it can be routed to the corresponding virtual host. I believe that this is a similar issue on OpenShift.

I hope this helps!

**Comment by Jeevan2000** [comment]
Hi, 
@tienhngnguyen @ashokkumarangadi does above solution works on OpenShift ??  Has anyone tried to solve it.
